mobile/lib/services/CommunicationService.dart

```diff
@@ -525,17 +527,22 @@ class CommunicationService {
     }
   }
 
-  Future<void> _handleEvent(dynamic event) async {
+  ///DON'T DO THIS!!! => Future<void> _handleEvent(dynamic event) async {
+  ///The jsonDecoding must be done before getting to this method
+  ///This ensures data is parse here in the correct format
+  Future<void> _handleEvent(Map<String, Object?> event) async {
     await communicationLock.synchronized(() async {
       print("Acquired communication lock for handling event.");
-      final Map<String, Object?> parsedEvent =
-          event is Map ? event : jsonDecode(event);
```

Comment:
event's type was dynamic to allow either String or Map. The event is then cast to a map if it is one, or decoded to a map if it is a String.